README.md

```diff
@@ -1,38 +1,36 @@
-# gitautosync
-
-Synchronizes a github repository with a local repository. Automatically deals with conflicts and produces useful output to stdout.
+# nbgitpuller
 
+One-way synchronization of a remote git repository to a local git repository,
+with automatic conflict resolution.
 
 # Installation
 
-Before installing, make sure you have a **Python 3** installation with
-the Jupyter environment installed.
-
-To install gitautosync, first install the package with `pip`. You should install
-the ``master`` branch of ``gitautosync``:
+There is no package on PyPI available right now. You can install directly from master:
 
     pip install git+https://github.com/data-8/gitautosync
 
-Next, enable the server extension with the following command:
-
-    jupyter serverextension enable --py nbgitautosync --sys-prefix
-
+You can then enable the serverextension
 
-# Usage
+    jupyter serverextension enable --py nbgitpuller --sys-prefix
 
-To use gitautosync, open the notebook. Depending whether you are working
-locally or via a JupyterHub, you'll use different syntax.
+# How it works
 
-If you're working off of a JupyterHub, use this syntax:
+nbgitpuller works by allowing you to construct a special URL, that when clicked
+will do the pulling for the user while rendering a nice status page. This is
+especially useful when running on a JupyterHub, since it allows easy distribution
+of materials to users without them having to understand git.
 
-    myjupyterhub.com/hub/user-redirect/git-sync?repo=<URL-to-my-repo>
+# Constructing the nbgitpuller URL
 
-If you're working locally, use this syntax:
+You can construct a working nbgitpuller URL like this:
 
-    localhost:8888/git-sync?repo=<URL-to-my-repo>
+```
+myjupyterhub.org/hub/user-redirect/git-sync?repo=<your-repo-url>&branch=<your-branch-name>&subPath=<subPath>
+```
 
-The repository will be cloned and placed in your root folder. If this
-repository already exists in your root folder, then files will be updated
-with any changes that exist in the repository from which we are pulling.
-If there are any merge conflicts, the user version will be kept, so your
-work won't be overwritten.
+- **repo** is the URL of the git repository you want to clone. This paramter is required.
+- **branch** is the branch name to use when cloning from the repository.
+  This parameter is optional and defaults to `master`.
+- **subPath** is the path of the directory / notebook inside the repo to launch after cloning.
```

Comment:
why not `filepath` for parity w/ binder?

Reply:
I might change that in a future patch :)